test/integration/ThreeChainEndToEnd.t.sol

```diff
@@ -0,0 +1,202 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.28;
+
+import "forge-std/Test.sol";
+
+import {IRecoverable} from "src/misc/interfaces/IRecoverable.sol";
+import {IERC20} from "src/misc/interfaces/IERC20.sol";
+
+import {IAdapter} from "src/common/interfaces/IAdapter.sol";
+import {IGateway} from "src/common/interfaces/IGateway.sol";
+import {IMultiAdapter} from "src/common/interfaces/adapters/IMultiAdapter.sol";
+
+import {ISpoke} from "src/spoke/interfaces/ISpoke.sol";
+
+import {IHub} from "src/hub/interfaces/IHub.sol";
+
+import "test/integration/EndToEnd.t.sol";
+
+enum CrossChainDirection {
+    WithIntermediaryHub, // C -> A -> B (Hub is on A)
+    FromHub, // C == A -> B (Hub is on C)
+    ToHub // C -> B == A (Hub is on B)
+
+}
+
+/// @title  Three Chain End-to-End Test
+/// @notice Extends the dual-chain setup to include a third chain (C) which acts as an additional spoke
+///         Hub is on Chain A, with spokes on Chains B and C
+///         C is considered the source chain, B the destination chain
+///         Depending on the cross chain direction, the hub is either on A or B or C
+contract ThreeChainEndToEndDeployment is EndToEndUtils {
+    using CastLib for *;
+    using MessageLib for *;
+
+    uint16 constant CENTRIFUGE_ID_C = 7;
+    ISafe immutable safeAdminC = ISafe(makeAddr("SafeAdminC"));
+
+    FullDeployer deployC = new FullDeployer();
+    LocalAdapter adapterCToA;
+    LocalAdapter adapterAToC;
+
+    CSpoke sC;
+    CSpoke sB;
+
+    function setUp() public override {
+        // Call the original setUp to set up chains A and B
+        super.setUp();
+
+        // Deploy the third chain (C)
+        adapterCToA = _deployChain(deployC, CENTRIFUGE_ID_C, CENTRIFUGE_ID_A, safeAdminC);
+        _setSpoke(deployC, CENTRIFUGE_ID_C, sC);
```

Comment:
This line is not needed, right? The spoke must be set under `setSpokes`

Reply:
Actually I think that's the issue your're having. We set sC to deployC, and setSpoke will avoid to reset that

Reply:
Ah thanks for that catch! I thought I removed it so I didn't check again.

Reply:
[02e12fd](https://github.com/centrifuge/protocol-v3/pull/415/commits/02e12fdc8b916497f127fc278558ea9511b64204)